src/corespotlight.cs

```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
 		bool Update (CSSearchableItemAttributeSet attributes, NSUrl contentUrl, [NullAllowed] out NSError error);
 	}
 
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQuery))]
+	[DisableDefaultCtor]
+	interface CSUserQuery
+	{
+		[Export ("initWithUserQueryString:queryContext:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string userQueryString, CSUserQueryContext queryContext);
+
+		[Export ("foundSuggestionCount")]
+		nint FoundSuggestionCount { get; }
+
+		[NullAllowed, Export ("foundSuggestionsHandler", ArgumentSemantic.Copy)]
+		Action<NSArray<CSSuggestion>> FoundSuggestionsHandler { get; set; }
+
+		[Export ("start")]
+		void Start ();
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQueryContext))]
+	interface CSUserQueryContext
+	{
+		[Static]
+		[Export ("userQueryContext")]
+		CSUserQueryContext UserQueryContext { get; }
+
+		[Static]
+		[Export ("userQueryContextWithCurrentSuggestion:")]
+		CSUserQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxSuggestionCount")]
+		nint MaxSuggestionCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSUserQueryContext))]
+	interface CSTopHitQueryContext
```

Comment:
There is a create method here, I would Disable the default ctor here

```suggestion
	[DisableDefaultCtor]
	interface CSTopHitQueryContext
```